composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/components/CalendarItem.kt

```diff
@@ -0,0 +1,228 @@
+    AppTheme {
+        Column {
+            CalenderItem(
+                "John Doe",
+                "Tesla",
+                painterResource(id = R.drawable.placeholder_profile),
+                1
+            )
+            CalenderItemPersonal(
+                "Cupra",
+                painterResource(id = R.drawable.placeholder_profile),
+                2
+            )
+            CalenderItem(
+                "Jane Doe",
+                "Tesla",
+                painterResource(id = R.drawable.placeholder_profile),
+                3
+            )
+        }
+    }
+}
+
+@Composable
+fun CalenderItemPersonal(
+    carBrand: String,
+    profilePic: Painter,
+    chargeTime: Int
+) {
+    val boxHeight = when (chargeTime) {
+        1 -> 56.dp
+        2 -> 112.dp
+        3 -> 168.dp
+        else -> 56.dp
+    }
+    val imageSize = when (chargeTime) {
+        1 -> 384.dp
+        2 -> 320.dp
+        3 -> 448.dp
+        else -> 384.dp
+    }
+    val imageRotation = when (chargeTime) {
+        1 -> - 30f
+        2 -> - 30f
+        3 -> 20f
+        else -> - 30f
+    }
+    val imageOffset = when (chargeTime) {
+        1 -> 150.dp to 80.dp
+        2 -> 50.dp to 100.dp
+        3 -> 20.dp to 100.dp
+        else -> 150.dp to 80.dp
+    }
+
+    Box(
+        modifier = Modifier
+            .size(312.dp, boxHeight)
+            .clip(RoundedCornerShape(4.dp))
+            .background(color = Colors.black)
+            .padding(vertical = 5.dp)
+    ) {
+        Image(
+            painterResource(id = R.drawable.rings),
+            contentDescription = null,
+            modifier = Modifier
+                .wrapContentSize(unbounded = true)
+                .size(imageSize)
+                .graphicsLayer {
+                    rotationZ = imageRotation
+                }
+                .offset(x = imageOffset.first, y = imageOffset.second)
+        )
+        Row(
+            modifier = Modifier.padding(horizontal = 12.dp, vertical = 8.dp)
+        ) {
+            Column {
+                Image(painter = profilePic, contentDescription = null)
+            }
+            Column(
+                modifier = Modifier.padding(start = 4.dp)
+            ) {
+                Text(
+                    text = SharedRes.strings.you.toStringResource(),
+                    color = Colors.white,
+                    style = TextStyles.calendar_item_name
+                )
+                Text(
+                    text = carBrand,
+                    color = Colors.mist,
+                    style = TextStyles.calendar_item_name
+                )
+            }
+            Spacer(modifier = Modifier.weight(1F))
+            Column {
+                Icon(
+                    imageVector = Icons.Default.ChevronRight,
+                    contentDescription = null,
+                    tint = Colors.white
+                )
+            }
+        }
+    }
+}
+
+@Composable
+fun CalenderItem(
```

Comment:
This is 90% duplicate code, I'm certain you can find a better solution for this.